src/main/java/seedu/address/model/lesson/StudentLessonInfo.java

```diff
@@ -0,0 +1,118 @@
+package seedu.address.model.lesson;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import seedu.address.model.student.Student;
+
+/**
+ * This class encapsulates an entry pairing Student objects with lesson info fields.
+ */
+public class StudentLessonInfo implements Serializable {
+    private final Student student;
+    private final boolean attendance;
+    private final int participationScore;
+
+    /**
+     * Creates a new StudentLessonInfo instance with the given parameters.
+     *
+     * @param student Student instance, must not be null.
+     * @param attendance True if the student has attended the lesson.
+     * @param participationScore An arbitrary integer describing the student's participation.
+     */
+    public StudentLessonInfo(Student student, boolean attendance, int participationScore) {
+        requireNonNull(student);
+        this.student = student;
+        this.attendance = attendance;
+        this.participationScore = participationScore;
+    }
+
+    /**
+     * Creates a new StudentLessonInfo with the default parameters.
+     * Attendance is set to false.
+     * Participation is set to 0.
+     *
+     * @param student The student instance to create a StudentLessonInfo for.
+     * @return A new StudentLessonInfo instance with default arguments.
+     */
+    public static StudentLessonInfo initialise(Student student) {
+        return new StudentLessonInfo(student, false, 0);
+    }
+
+    /**
+     * Returns true if this StudentLessonInfo instance belongs to the given student.
+     * Uses weak equality checking (name equality) since students should be unique by name.
+     *
+     * @param student The student instance to check against.
+     * @return True if this instance belongs to the given student.
+     */
+    public boolean isForStudent(Student student) {
```

Comment:
The Method's Name is slightly unclear, but I see that isSameStudent(Student) is already a method for the Student Class.